### DISCUSSION TITLE
Drag and Drop files order

### DISCUSSION THREAD

**Discussion by renrutsirhc** [AvaloniaUI/Avalonia#7160] [Q&A]
Hi all,

I'm dragging and dropping files from windows explorer into my application where they are added to a listbox. All working just fine except for one thing. Let's say I have 5 files named:

- 1.pdf
- 2.pdf
- 3.pdf
- 4.pdf
- 5.pdf

I select them all and initiate a drag from windows explorer with mouse down over 1.pdf. Upon droppping, over my listbox, the DragEventArgs e.Data.GetFileNames() has the 5 files in the correct order.

If I now repeat the drag but this time initiating the drag with the mouse down over 3.pdf, the order of the files is now:

- 3.pdf
- 4.pdf
- 5.pdf
- 1.pdf
- 2.pdf

Is this by design? My instinct is that it would be more intuitive if the order of the files in the DragEventArgs was as they are ordered in Windows Explorer irrespective of which is 'clicked' when initiating the drag.

Thanks for your help!
Chris

**Comment by MarkusKgit** [comment]
I just checked with two other applications (PDF24 and Paint.Net) and they behave the same
